app/src/androidTest/java/com/example/android/justjava/Models/JustJavaModel.kt

```diff
@@ -0,0 +1,214 @@
+package com.example.android.justjava.Models
+
+import android.support.test.espresso.ViewInteraction
+
+import com.example.android.justjava.Helpers.Matchers.ToastMatcher
+import com.example.android.justjava.R
+
+import android.support.test.espresso.Espresso.closeSoftKeyboard
+import android.support.test.espresso.Espresso.onView
+import android.support.test.espresso.action.ViewActions.click
+import android.support.test.espresso.action.ViewActions.replaceText
+import android.support.test.espresso.assertion.ViewAssertions.matches
+import android.support.test.espresso.matcher.ViewMatchers.isChecked
+import android.support.test.espresso.matcher.ViewMatchers.isDisplayed
+import android.support.test.espresso.matcher.ViewMatchers.isNotChecked
+import android.support.test.espresso.matcher.ViewMatchers.withId
+import android.support.test.espresso.matcher.ViewMatchers.withText
+import com.example.android.justjava.TestData.TestStrings.firstNameOnly
+import com.example.android.justjava.TestData.TestStrings.longName
+import com.example.android.justjava.TestData.TestStrings.order1
+import com.example.android.justjava.TestData.TestStrings.order2
+import com.example.android.justjava.TestData.TestStrings.shortenedName
+import com.example.android.justjava.TestData.TestStrings.testName
+import org.hamcrest.Matchers.containsString
+
+class JustJavaModel {
+    private val nameEditText = onView(withId(R.id.name_field))
+    private val toppingsHeader = onView(withId(R.id.toppings))
+    private val creamCheckBox = onView(withId(R.id.whipped_cream_checkbox))
+    private val chocolateCheckBox = onView(withId(R.id.chocolate_checkbox))
+    private val quantityHeader = onView(withId(R.id.quantity))
+    private val minusButton = onView(withId(R.id.minusButton))
+    private val quantityDisplay = onView(withId(R.id.quantity_number_text_view))
+    private val plusButton = onView(withId(R.id.plusButton))
+    private val summaryHeader = onView(withId(R.id.orderSummaryHeader))
+    private val totalAmount = onView(withId(R.id.order_summary_text_view))
+    private val orderButton = onView(withId(R.id.orderButton))
+
+    // add name testName to the name textEdit field
+    private fun addNameToEditText() {
+        nameEditText.perform(replaceText(testName))
+    }
+
+    // verify correct name was added to the field
+    private fun verifyNameAdded() {
+        nameEditText.check(matches(withText(testName)))
+    }
+
+    // add name testName to field and verify
+    fun addNameAndVerify() {
+        addNameToEditText()
+        verifyNameAdded()
+    }
+
+    // add a name greater than character limit set in activity_main.xml
+    private fun exceedCharLimit() {
```

Comment:
You have a mixture of private and public functions listed here. Not sure if I mentioned it but as per our coding standards, the private functions should be at the bottom below the public functions. I think that is pretty standard anyways. Maybe just move the private functions at the bottom.